### DISCUSSION TITLE
So net::ERR_ADDRESS_UNREACHABLE after upgrade to 15.2 w/ 0.12.1

### DISCUSSION THREAD

**Discussion by jboxman** [jeffvli/feishin#872] [General]
Same setup as before, so I deleted the _Application Support_ directory, deleted from _Applications_, and installed fresh. This time, I'm able to get Developer Tools open, and I'm seeing a socket error.

This is puzzling. I'm streaming from the Navidrome server right now in a regular Web browser. Target is:

`http://<server name>:4533/users/authenticatebyname`

Hostname or IP address, makes no different. Unreachable.

Out of ideas.

**Comment by jeffvli** [comment]
The target URL is incorrect. When you add your server, you should only be entering `http://<server name>:4533` without anything after that.

**Reply by jboxman** [reply]
@jeffvli, yes, exactly.

My original configuration came over from upgrading, and that is the form that I used. After reinstalling and deleting all configuration files, and starting with a brand new, empty **Manage Servers** modal, I put in the URL as before, just `http://<server name>:4533` as you've indicated.

The URL error I get back from the console is from an HTTP request that's sent out when I hit **Add** after entering the server details. I'm not familiar with the Navidrome protocol, so I don't know what it means. In previous versions, this all worked. I assume this is somehow related to OS X 15.2, but I haven't gotten any new notifications to enable permissions.

Several other applications (browsers) did prompt to enable access to the _Local Network_ (a new? security section entry), so I think this could be related to a new security permission that Apple requires?

I don't know how to confirm this. I'm not sure if it is possible to manually give a program access to some security zone. I've always been prompted by OS X about it.

**Reply by jeffvli** [reply]
Did you make sure to change server selector to `Navidrome` in the Add Server window? 

![image](https://github.com/user-attachments/assets/2316ca89-03b3-4d0b-a77c-d34c98d35770)

`authenticatebyname` is only used by Jellyfin, which is probably why you're seeing the error.

**Reply by jboxman** [reply]
@jeffvli, oops, I forgot about that. So instead I get an error for the correct Navidrome API endpoint, after making the correct selection.

```
Failed to load resource: net::ERR_ADDRESS_UNREACHABLE
http://<me>:4533/auth/login
```

**Reply by jboxman** [reply]
And it works in the browser and from the CLI:

```
curl -v http://me:4533/auth/login
* Host me:4533 was resolved.
* IPv6: (none)
* IPv4: 192.168.70.3
*   Trying 192.168.70.3:4533...
* Connected to me (192.168.70.3) port 4533
> GET /auth/login HTTP/1.1
> Host: me:4533
> User-Agent: curl/8.7.1
> Accept: */*
>
* Request completely sent off
< HTTP/1.1 405 Method Not Allowed
< Allow: POST
< Permissions-Policy: autoplay=(), camera=(), microphone=(), usb=()
< Referrer-Policy: same-origin
< Vary: Origin
< X-Content-Type-Options: nosniff
< X-Frame-Options: DENY
< Date: Fri, 20 Dec 2024 04:46:50 GMT
< Content-Length: 0
<
* Connection #0 to host me left intact
* ```
*

**Reply by jeffvli** [reply]
Maybe try the solution listed [in this reddit post](https://www.reddit.com/r/PleX/comments/1gm8pm4/solution_for_err_address_unreachable_in_macos/). I don't have a macOS device so this issue is foreign to me.

**Reply by jboxman** [reply]
Sadly Feishin does not show up in the list. There's no option to _add_ it.

This doesn't look promising, though: https://developer.apple.com/documentation/technotes/tn3179-understanding-local-network-privacy

Yep: https://apple.stackexchange.com/questions/475737/how-can-i-manually-allow-local-network-access-for-an-app

So, shouldn't have upgraded OS X. Sigh. I mean maybe I can do something weird like run it through a Docker container or whatever, but that kind of defeats the purpose. Maybe a local forwarding proxy, but ugh.

**Reply by jeffvli** [reply]
Seems like I'll have to try to code-sign the app for OSX.

**Reply by jboxman** [reply]
Works with an `ssh` tunnel, so definitely OS X.
